### PR TITLE
Refactor modals. Issue: #2645

### PR DESCRIPTION
**Pull request by garethbowen**

### CODE REVIEW

File: static/js/controllers/edit-user.js [hunk 1]
```diff
@@ -172,37 +162,25 @@ var modal = require('../modules/modal');
...
         $scope.errors = {};
+        $scope.setProcessing();
         if (validatePassword()) {
-          var pane = modal.start($('#update-password'));
           var updates = { password: $scope.editUserModel.password };
           UpdateUser($scope.editUserModel.id, null, updates)
             .then(function() {
-              updateComplete(pane);
+              $scope.setFinished();
+              $window.location.reload(true);
```
Comment (estellecomment) [inline static/js/controllers/edit-user.js]: you're not broadcasting `UsersUpdated` here. Is it on purpose?

File: static/js/directives/modal.js [hunk 2]
```diff
@@ -1,17 +1,38 @@
+/**
+ * Directive for wrapping modal bodies with the modal chrome.
```
Comment (estellecomment) [inline static/js/directives/modal.js]: it's the chrome browser, made out of modals? :brain: :boom: 🇬🇧 
"Directive for boilerplate for modal bodies"

Comment (garethbowen) [inline static/js/directives/modal.js]: Hah. Lower case "chrome"  :P

File: static/js/controllers/edit-user.js (hunk 1 above)
Comment (garethbowen) [inline static/js/controllers/edit-user.js]: Yes. When updating your own password the page gets refreshed so we don't need to broadcast that the users have changed.

**Pull request merged**